modules/inputSpinner/InputSpinner.tsx

```diff
@@ -0,0 +1,157 @@
+    boxSizing: "border-box",
+    margin: 0,
+    "@selectors": {
+      "&:hover :not(.disabled).value": {}
+    }
+  },
+  button: {
+    outline: "none",
+    cursor: "pointer",
+    "@selectors": {
+      "&.disabled": {
+        cursor: "initial"
+      }
+    }
+  },
+  buttonMin: {},
+  buttonMax: {},
+  value: {
+    alignItems: "center",
+    justifyContent: "center",
+    minWidth: 21,
+    ...theme.typography.body,
+    "@selectors": {
+      "&::-webkit-inner-spin-button": {
+        "-webkit-appearance": "none",
+        "-moz-appearance": "none",
+        appearance: "none",
+        margin: 0
+      },
+      "&::-webkit-outer-spin-button": {
+        "-webkit-appearance": "none",
+        "-moz-appearance": "none",
+        appearance: "none",
+        margin: 0
+      },
+      "&.disabled": {},
+      "&.focus:not(.disabled)": {}
+    }
+  }
+});
+
+const InputSpinner: React.FC<IProps & WithStylesProps> = ({
+  styles,
+  cx,
+  className,
+  value,
+  onChange,
+  parentStylesheet,
+  ...props
+}) => {
+  const { min, max, step, disabled } = props;
+  const [_value, setValue] = useState(0);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const containerStyle = cx("container", styles.container, className);
+  const buttonStyle = cx("button", styles.button, disabled && "disabled");
+  const valueStyle = cx(
+    "value",
+    styles.value,
+    disabled && "disabled",
+    isFocused && "focus"
+  );
+
+  useEffect(() => {
+    if (value) {
+      setValue(value);
+    }
+  }, [value]);
+
+  const fixedValue = useMemo(() => {
+    const valueState = value ?? _value;
+    if (!valueState) {
+      return "00";
+    }
+    if (valueState < 10 && valueState > 0) {
+      return `0${valueState}`;
+    }
+    return valueState;
+  }, [_value, value]);
+
+  const changeValue = useCallback(
+    newValue => {
+      if (disabled) return;
+
+      setValue(newValue);
+      return onChange?.(newValue);
+    },
+    [onChange, disabled]
+  );
+
+  const increase = useCallback(() => {
+    const newValue = max
+      ? Math.min(max, _value + (step ?? 1))
+      : _value + (step ?? 1);
+    changeValue(newValue);
+  }, [_value, step, max, changeValue]);
+
+  const decrease = useCallback(() => {
+    const newValue = min
+      ? Math.max(min, _value - (step ?? 1))
+      : _value - (step ?? 1);
+    changeValue(newValue);
+  }, [_value, step, min, changeValue]);
+
+  return (
+    <div className={containerStyle}>
+      <button
+        type="button"
+        className={cx(buttonStyle, styles.buttonMin)}
+        onClick={decrease}
+        onBlur={() => setIsFocused(false)}
+        onFocus={() => setIsFocused(true)}
+      >
+        <ButtonText>-</ButtonText>
+      </button>
+      <input
+        {...props}
+        className={valueStyle}
+        type="number"
+        value={fixedValue}
+        onChange={e => changeValue(parseInt(e.target.value, 10))}
+        style={{ width: `${fixedValue.toString().length * 10}px` }}
```

Comment:
This is very restrictive as far as custom styles go, no?

Reply:
Yes, this is a problem. There's no reasonable way to change the width of an input according to its content. We could fix it as the designers want it but it would be wrong when reaching 3 digit numbers.  What's your recommendation?

Reply:
It seems like you have already looked into this issue and there isn't a simple solution. 
Thus, I suggest leaving it as is and if we run into any issue in the future we can revisit.
Let's not solve problems we are not sure we will have 😄